src/network/wait-online/wait-online.c

```diff
@@ -48,6 +49,7 @@ static int help(void) {
                "  -6 --ipv6                 Requires at least one IPv6 address\n"
                "     --any                  Wait until at least one of the interfaces is online\n"
                "     --timeout=SECS         Maximum time to wait for network connectivity\n"
+               "     --dns                  Requires at least one DNS server to be accessible\n"
```

Comment:
I'd prefer if this was an "extended boolean" enum. i.e. that parses as boolean, but also can take an additional value for disabling the connectivity check (because I think it's problematic in some environments, because resolved might be allow to talk to DNS while other stuff might not be). Maybe call that extra value "passive" or so?

i.e. `--dns=yes`, `--dns=no`, `--dns=passive`

this is easy to do via DEFINE_PRIVATE_STRING_TABLE_LOOKUP_FROM_STRING_WITH_BOOLEAN() and related boilerplate macros

Reply:
I guess this is no longer needed when doing the check from resolved, right?

Reply:
dunno, it'd at least make it take a boolean for now, so that we can extend it later.

i.e. `--dns=yes`. we have a simpler parser for that parse_boolean_argument()